### PR TITLE
find active and inactive preferences of an experience

### PR DESCRIPTION
**Pull request by CarLosVegga**
Hi everyone!

Some context of my PR:

Previously I had an endpoint for each active and inactive preferences.
I decided to merge those two into a single one.

I found a bug while dealing with this PR. It seems like I was creating an Experience until the user firsts uploads the data (which at the beginning made sense). However, my form to create an experience is next to the select preferences to be rated by adventurers. 
image

So, if the user didn't follow the expected user flow (first submit your experience information and then modify your preferred feedback) the second form would simply die (because it looks for your activePreferences array, which is instantiated to be null once you submit your experience info!)

I decided to take an approach like the one I followed with the UserPreference: instantiate it once the user signs up.
That is why I modified **visitor.jsx**

However, on the Adventurer side that killed the rendering of restaurants. Since now I am creating an experience with no data, and my algorithm finds the distance of your current location and that one registered in the db, it was calling a nonexistent value. To solve that, I simply added a continue if there is no address on **routes/adventure.js**
I think I should implement later on a better checking method. So far the way I structured it, it can't upload a preference through the UI if a mandatory field is not complete, so for this delivery I think I will leave it that way.

I also modified the update router request on **routes/adventure.js**, but I will refactor that as my next step. The reason behind that is that I noticed I was repeating over and over the toast method. Most of the time I would call it twice, once for success and another for error, and I was repeating the error messages in most of the cases. 
To avoid that, I decided it would be also a good idea to have a module control it. So, if I ever want to make a general change like the display time, I can easily do so.

Thanks for any comments on this!

### CODE REVIEW

File: capstone/src/components/User/Experience/Experience.jsx [hunk 1]
```diff
@@ -42,10 +42,10 @@ export default function Experience({ setIsLoggedIn, isLoggedIn }) {
     } else if (username != null) {
       Promise.all([getExperienceInfo(username), getAllFeedbackInfo()]).then(
         ([expRes, feedbackRes]) => {
-          // If there are no records on the experience that the user owns
+          // If there are no values for a field
           // send to that page to promote submission.
           // Otherwise, just set values in state
-          if (Object.keys(expRes.data).length === 0) {
+          if (Object.keys(expRes.data).every((experienceField) => !!experienceField)) {
```
Comment (vqdo) [inline capstone/src/components/User/Experience/Experience.jsx:48]: Is `activeFeedback` one of the experience fields? I noticed you initialized this to an array above. If so, arrays are always [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy). I

File: capstone/src/components/User/Toast.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Show toast value
+export default function callToast({ toast, title, description, status }) {
```
Comment (vqdo) [inline capstone/src/components/User/Toast.jsx:2]: Personally I think it's a bit confusing to pass in the toast object. I would recommend something like:

```
function getToastOptions({title, description, status}) {
  if (status === 'error') {
    // return other toast options
    return {...}
  }
  return {
    title,
    description,
  // You can pass in defaults here
    status: status ?? 'success',
  }
}
```

```
const toast = useToast();
toast(getToastOptions({ /* options */ }));
```

**Review by vqdo — APPROVED** [review]

File: capstone/src/components/User/Experience/Experience.jsx (hunk 1 above)
Comment (CarLosVegga) [reply capstone/src/components/User/Experience/Experience.jsx:48]: Wow! I definitely didn't notice that!!!
I just added that to an issue I plan to address soon
Thank you so much for that one

File: capstone/src/components/User/Toast.jsx (hunk 2 above)
Comment (CarLosVegga) [reply capstone/src/components/User/Toast.jsx:2]: That actually makes so much sense! I just included that change, thanks for that!

**Pull request merged**